### PR TITLE
Add CI workflows to repo

### PR DESCRIPTION
**Pull request by eddyleelin**
This PR adds continuous integration (CI) support via GitHub Actions. The following workflows have been adapted primarily from the Open-Telemetry/OpenTelemetry-Go repo:

* [CodeQL Analysis](https://github.com/open-telemetry/opentelemetry-go/actions/workflows/codeql-analysis.yml) triggered once per day
* [Dependabot-Tidier](https://github.com/open-telemetry/opentelemetry-go/actions/workflows/dependabot.yml) triggered upon the logic `if contains(github.event.pull_request.labels.*.name, 'dependencies')` 
* [Run Gosec](https://github.com/open-telemetry/opentelemetry-go/actions/workflows/gosec.yml) triggered once per day
* [ci](https://github.com/open-telemetry/opentelemetry-go/actions/workflows/ci.yml) triggered upon PR
* [markdown](https://github.com/open-telemetry/opentelemetry-go/actions/workflows/markdown.yml) triggered upon PR

**How has this change been tested?**
As can be seen below in the Checks section, the semverconv module is being tested with the new workflows. Currently, the `ci / test-coverage` test fails, as this repo has not yet been added to Codecov. To add it, a maintainer could follow this link: https://app.codecov.io/gh/open-telemetry/opentelemetry-go-build-tools/

The CI workflows have been integrated into a separate fork (with codecov integrated) and tested against an empty commit, and all checks passed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+TOOLS = $(CURDIR)/.tools
+
+$(TOOLS):
+	@mkdir -p $@
+$(TOOLS)/%: | $(TOOLS)
+	cd $(TOOLS_MOD_DIR) && \
+	$(GO) build -o $@ $(PACKAGE)
+
+SEMCONVGEN = $(TOOLS)/semconv-gen
+$(TOOLS)/semconv-gen: PACKAGE=go.opentelemetry.io/build-tools/semconvgen
```
Comment (Aneurysm9) [inline Makefile]:
```suggestion
$(TOOLS)/semconv-gen: PACKAGE=go.opentelemetry.io/opentelemetry-go-build-tools/semconvgen
```

This probably isn't needed at all though.

**Review by Aneurysm9 — APPROVED** [review]
I think the codecov upload is failing because the open-o11y org isn't set up there.  Let's get this upstream and it should Just Work™.

File: Makefile (hunk 1 above)
Comment (eddyleelin) [reply Makefile]: Removed it, thanks

**Comment by eddyleelin** [comment]
Keeping this open to show how the checks look until the upstream PR is merged.

**Pull request closed without merging**